BrainPortal/app/views/tool_configs/_form_fields.html.erb

```diff
@@ -22,172 +21,187 @@
 #
 -%>
 
-<div class="groupentry">
 
-  <%= hidden_field_tag "tool_config[tool_id]",     @tool_config.tool_id %>
-  <%= hidden_field_tag "tool_config[bourreau_id]", @tool_config.bourreau_id %>
 
+<%= error_messages_for @tool_config %>
+
+<%= show_table_context(@tool_config) do |cf| %>
   <% if @tool_config.tool_id && @tool_config.bourreau_id %>
-  <span title="Project access for this configuration.">
-    <p><%= f.label :group_id, "Available to members of project:" %>
-    <%= group_select "tool_config[group_id]", @tool_config.group_id %>
-    </p>
-  </span>
-
-  <span title="Version.">
-    <p><%= f.label :version_name, "Version:" %>
-    <%= f.text_field :version_name %><br/>
-    <div class="field_explanation">Must be a simple string that represent a short identifier for the version. First character must be alphanum, and can contain only alphanums, '.', '-', '_', ':' and '@'</div>
-    </p>
-  </span>
-
-  <span title="Brief description for this configuration.">
-    <p><%= f.label :description, "Description:" %><br/>
-    <%= f.text_area :description, :rows => 10, :cols => 40 %><br/>
-    <div class="field_explanation">The first line must be a short summary, and the rest are for any special notes for the users.</div>
-    </p>
-  </span>
-
-  <span title="Suggested number of CPUs to use in parallel, per task">
-    <p><%= f.label :ncpus, "Suggested number of CPUs to use in parallel, per task:" %>
-    <%= f.select :ncpus, [ 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 14, 16, 20, 24, 32, 36, 40, 48, 56, 64, 128, 192, 256, 320, 384, 512 ] %>
-    </p>
-  </span>
+    <%= show_table(@tool_config,  :form_helper => cf, :header => "Version configuration info",
+                                 :edit_condition => check_role(:admin_user)) do |t| %>
+
+      <% t.edit_cell(:group_id, :header => "Project access",
+                     :content => link_to_group_if_accessible(@tool_config.group_id),
+                     :td_options => {:title => "The version is available to members of project"}) do %>
+        <%= group_select "tool_config[group_id]", @tool_config.group_id %>
+        <%= hidden_field_tag "tool_config[tool_id]", @tool_config.tool_id %>
```

Comment:
Move these two hideen params outside the show table and remove from all other show tables